cmds/core/ts/ts.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// ts prepends each line of stdin with a timestamp.
+//
+// Synopsis:
+//     ts
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+	"time"
+
+	flag "github.com/spf13/pflag"
+	"github.com/u-root/u-root/pkg/ts"
+	"golang.org/x/text/transform"
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 0 {
+		log.Fatal("Expected no arguments")
```

Comment:
ah man. Can you have this be
Usage: ts
and leave it at that?

Reply:
the complexity of using transform, vs. the problems you report, make me wonder if the additional mess from using it (and even creating a package, yet) is worth it, vs. a readline/writeline loop right here in this program that prints a stamp.

Reply:
readline is not good in this case because it forces the input to be lined buffered. I've seen plenty of code which performs an expensive operation mid-line. For example, something line:

```
printf("Waiting for signal...");
someSlowOperation(); // can take seconds
printf("DONE\n");
```

If I used readline, the output from the first printf would be stuck in some internal buffer and not visible until after the second printf. This is not so great because you would like to see debug output as soon as possible.

The moreutils Perl version of `ts` has this problem. It also makes the mistake of measuring when the newline was printed at the end of line and printing the timestamp at the beginning of the line. For example:

```
$ (printf "Initializing kernel\n"; sleep 1; printf "Waiting for signal..."; sleep 1; printf "DONE\n"; echo Success) | ts %.s
1583734537.887905 Initializing kernel
1583734539.866401 Waiting for signal...DONE
1583734539.866562 Success
```
Notice how there incorrectly appears to be 2 seconds between printing "Initializing kernel" and printing "Waiting for signal".

The correct way is to print the timestamp upon reading the first character immediately after a newline (and at the beginning of input). The Go version I pushed does this correctly:

```
$ (printf "Initializing kernel\n"; sleep 1; printf "Waiting for signal..."; sleep 1; printf "DONE\n"; echo Success) | ./ts
[0.0000s] Initializing kernel
[1.0004s] Waiting for signal...DONE
[2.0032s] Success
```

Unlike the Perl version, this Go version has the nice property: **The timestamp's value is the time when it was printed.**